iwyu_ast_util.cc

```diff
@@ -1412,4 +1430,13 @@ TemplateArgumentListInfo GetExplicitTplArgs(const Expr* expr) {
   return explicit_tpl_args;
 }
 
+bool IsOverriddenMethodDecl(const clang::Decl* decl) {
+  if (const auto* method = dyn_cast_or_null<CXXMethodDecl>(decl)) {
+    return (!method->isThisDeclarationADefinition() &&
```

Comment:
I don't think this belongs here. There could be cases where knowing that a decl is an override is useful whether it has a body or not.

I think it should be moved to `IsPartOfOverrideDecl` where it could be made more specific to handle incomplete return types correctly, instead of bailing out completely.

If I am not mistaken, an inline override with unused arguments would need an include for them, as it is now:
```c++
class Override: public Base {
  void MyFunc(A& /*unused*/) override { // <-- need to include (forward-declare) A
  }
};
```

Reply:
> I don't think this belongs here.

Quite right! And now that this method was only called from one place, I decided to inline it completely.

> If I am not mistaken, an inline override with unused arguments would need an include for
> them, as it is now:

That does look a little weird; maybe the special casing for definitions should only apply to the return type. I need to double-check what the compiler thinks about that one.

Reply:
I've spent the past few days trying to special case the return type of the function definition, but I haven't found anything that works well yet.

The challenge is that there's no obvious way to traverse the ASTNodes downwards after traversal. So phrasing the "ignore the return type of a user reported for CXXMethodDecl" logic is surprisingly hard.

I have some ideas for how to handle this better, but it requires larger-scale changes that we should be able to use for other "backward lookups", so I'd rather do that as a separate change altogether. 

OK to leave this so it requires a decl when unused args are part of the definition signature?

Reply:
Yes, sure, I am fine with it. I thought it could be as simple as getting the return from the method decl and compare against the ast node, but I am likely missing something.

Reply:
I've tried 12 variations on that theme without getting to something that works reliably :)